### PR TITLE
feat(indexer): Add metrics to indexer/api

### PR DESCRIPTION
**Pull request by roninjin10**
- Add metrics to api

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #7037** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7037" target="_blank">Graphite</a> 
    * **PR #7038** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7038" target="_blank">Graphite</a>  👈
      * **PR #7040** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7040" target="_blank">Graphite</a> 
      * **PR #7039** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7039" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7038?utm_source=stack-comment).

File: indexer/api/api.go [hunk 1]
```diff
@@ -43,3 +59,17 @@ func (a *Api) Listen(ctx context.Context, port int) error {
 
 	return err
 }
+
+func (a *Api) ListenMetrics(ctx context.Context, port int) error {
+	a.log.Info("metrics server listening...", "port", port)
+
+	server := http.Server{Addr: fmt.Sprintf(":%d", port), Handler: a.MetricsRouter}
+	err := httputil.ListenAndServeContext(ctx, &server)
```
Comment (roninjin10) [inline indexer/api/api.go:98]: ListenAndServe on metrics instead

File: indexer/api/api.go [hunk 2]
```diff
@@ -3,43 +3,114 @@ package api
...
-	return &Api{log: logger, Router: r}
+	return err
 }
 
-func (a *Api) Listen(ctx context.Context, port int) error {
-	a.log.Info("api server listening...", "port", port)
-	server := http.Server{Addr: fmt.Sprintf(":%d", port), Handler: a.Router}
+func (a *Api) startServer(ctx context.Context) error {
+	a.log.Info("api server listening...", "port", a.apiConfig.Port)
+	server := http.Server{Addr: fmt.Sprintf(":%d", a.apiConfig.Port), Handler: a.Router}
```
Comment (roninjin10) [inline indexer/api/api.go]: Prefer to pass in the port when listen is called but doesn't really matter much in this case and this allowed me to copy paste `runProcess`

**Review by hamdiallam — APPROVED** [review]

File: indexer/api/api.go [hunk 3]
```diff
@@ -4,42 +4,113 @@ import (
...
 }
 
-func NewApi(logger log.Logger, bv database.BridgeTransfersView) *Api {
-	r := chi.NewRouter()
-	h := routes.NewRoutes(logger, bv, r)
+const (
+	MetricsNamespace = "op_indexer"
+)
+
+func ChiMetricsMiddleware(rec metrics.HTTPRecorder) func(http.Handler) http.Handler {
```
Comment (hamdiallam) [inline indexer/api/api.go]: ffti: this doesn't have to be exported

Comment (roninjin10) [reply indexer/api/api.go]:
```suggestion
func chiMetricsMiddleware(rec metrics.HTTPRecorder) func(http.Handler) http.Handler {
```

**Pull request merged**